### PR TITLE
Converted OKE run to use JenkinFile

### PR DESCRIPTION
**Pull request by marinakog**
https://build.weblogick8s.org:8443/job/wko-oke-test-seqential/11
after change is merged will convert to use main as default branch

### CODE REVIEW

File: JenkinsOKEfile [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+                     description: 'Collect logs for successful runs. Default is false.',
+                     defaultValue: false
+        )
+    }
+
+    stages {
+        stage('Filter unwanted branches') {
+            when {
+                anyOf {
+                    branch 'okejen'
```
Comment (robertpatrick) [inline JenkinsOKEfile:155]: This is almost certainly not what you want to checkin

File: JenkinsOKEfile [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
+                        DOCKERHUB_USERNAME = credentials("${dockerhub_username_creds}")
+                        DOCKERHUB_PASSWORD = credentials("${dockerhub_password_creds}")
+                    }
+                    steps {
+                        sh 'echo ${DOCKERHUB_PASSWORD} | docker login -u ${DOCKERHUB_USERNAME} --password-stdin'
+                        sh "mvn -DtrimStackTrace=false clean install"
+                    }
+                }
+
+                stage('Run Sonar Analysis') {
```
Comment (robertpatrick) [inline JenkinsOKEfile:288]: You should not run Sonar on these builds.  Sonar is already running on the PR gate builds, which should be sufficient.

File: JenkinsOKEfile [hunk 3]
```diff
@@ -0,0 +1,594 @@
...
+                            echo "${WORKSPACE}/.mvn/maven.config contents:"
+                            cat "${WORKSPACE}/.mvn/maven.config"
+                        '''
+                        withSonarQubeEnv('SonarCloud') {
+                            sh "mvn sonar:sonar"
+                        }
+                    }
+                }
+
+                 stage('Verify Sonar Quality Gate') {
```
Comment (robertpatrick) [inline JenkinsOKEfile:313]: Same comment…don’t run Sonar

File: JenkinsOKEfile [hunk 4]
```diff
@@ -0,0 +1,594 @@
...
+                        sh "mkdir -m777 -p ${WORKSPACE}/bin"
+                    }
+                }
+                stage ('Install oci cli') {
+                    environment {
+                        runtime_path = "${WORKSPACE}/bin:${PATH}:/bin"
+                    }
+                    steps {
+                        sh '''
+                            export PATH=${runtime_path}
```
Comment (robertpatrick) [inline JenkinsOKEfile:333]: Installing/configuring software at a global level is a really unfriendly thing to do in a shared environment.   You should consider another way to handle this.

**Review by robertpatrick — CHANGES_REQUESTED** [review]
Please review my comments.

please rename the file to Jenkinsfile.okd

File: JenkinsOKEfile (hunk 4 above)
Comment (marinakog) [reply JenkinsOKEfile]: fixed all comments, waiting to get python3 to be part of the oci image creation. Meanwhile everything else is ready for review. Moved to single branch pipeline : https://build.weblogick8s.org:8443/job/wko-oke-test-seqential/11/

**Review by robertpatrick — APPROVED** [review]

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**